extensions/mssql/src/ui/scriptableDialogBase.ts

```diff
@@ -63,13 +63,16 @@ export abstract class ScriptableDialogBase<OptionsType extends ScriptableDialogO
 	protected abstract get isDirty(): boolean;
 
 	protected override onFormFieldChange(): void {
-		this._scriptButton.enabled = this.isDirty;
+		this._scriptButton.enabled = this.isDirty && this.modelView.valid;
 		this.dialogObject.okButton.enabled = this.isDirty;
 	}
 
 	protected override async initialize(): Promise<void> {
 		await this.initializeData();
 		await this.initializeUI();
+		this.disposables.push(this.modelView.onValidityChanged((isValid: boolean) => {
+			this._scriptButton.enabled = isValid;
```

Comment:
Shouldn't this also be `this._scriptButton.enabled = this.isDirty && this.modelView.valid;`?

It might be better to have an `updateScriptEnabledState` helper function that is called in both scenarios.